src/LinuxTracing/LeafFunctionCallManagerTest.cpp

```diff
@@ -173,9 +174,9 @@ TEST_F(LeafFunctionCallManagerTest, PatchCallerOfLeafFunctionReturnsErrorOnTooSm
   callchain.push_back(kTargetAddress3 + 1);
 
   CallchainSamplePerfEventData event_data = BuildFakeCallchainSamplePerfEventData(callchain);
-  event_data.regs->bp = 2 * kStackDumpSize;
-  event_data.regs->sp = 0;
-  event_data.regs->ip = kTargetAddress1;
+  event_data.regs.get()[6] = 2 * kStackDumpSize;
```

Comment:
When addressed the comment above: move those assignment s up to the `ip` one.

Reply:
You mean to make the assignment similar to the assignment in your first comment? Create a struct then copy it using memcpy?

Reply:
I think in this case, it is even the exact same "regs" as we create before. But yes, the idea would be:
1. create a readable struct for the registers
2. fill-in all the data that we want to set during the test
3. copy the content of this struct into the perf event

Reply:
Should be done everywhere.